src/Storages/Distributed/DirectoryMonitor.cpp

```diff
@@ -262,6 +267,14 @@ namespace
             return;
         }
 
+        /// If connection does not use compression, we have to uncompress the data.
+        if (!compression_expected)
+        {
+            writeAndConvert(remote, in);
```

Comment:
Wroth adding a log message?

Plus maybe also a test, but seems that this will require integration test...

Reply:
> Wroth adding a log message?

I think that this branch represents normal (non exceptional) use case. Because compression can be disabled for remote host manually. It even has its use cases, e.g. 100 Gbit network. It just did not work before and nobody argued :)

> Plus maybe also a test, but seems that this will require integration test...

Existing tests have failed, and this was the only motivation adding this support.
So, we already have tests for it.

Reply:
Ah, you mean the tests for no-compression by default for localhost.

Reply:
>Existing tests have failed, and this was the only motivation adding this support.

I see, i.e. INSERT with prefer_localhost_replica=0 in 01675_distributed_bytes_to_delay_insert

> Ah, you mean the tests for no-compression by default for localhost.

It is not that easy I guess, perf test looks enough to me.

> I think that this branch represents normal (non exceptional) use case.

Agree.

> Because compression can be disabled for remote host manually in cluster config. It even has its use cases, e.g. 100 Gbit network. It just did not work before and nobody argued :)

Actually decompression will be done anyway, the only difference is who will do it.

Reply:
> Actually decompression will be done anyway

Ok. I mean distributed SELECT queries.